### DISCUSSION TITLE
Tree View

### DISCUSSION THREAD

**Discussion by jd-isc** [directus/directus#5546] [Archive]
Are there instructions for creating a tree view in a data model? I would like to display the O2M PK collection fields with a list of the FK collections data below. Also, with the ability to click on the FK collection which will allow an edit or additions. Is this possible?

**Comment by rijkvanzanten** [comment]
@jd-isc The current tree-view interface works based on a recursive o2m configuration. So assuming you're talking about parent->children within the same collection, this is possible yeah 👍🏻

**Reply by jd-isc** [reply]
@rijkvanzanten great, thank you. Are there instructions on how to set up this tree view properly? Although I think I have the set up correct, I am getting an error 'The tree view interface only works for recursive relationships'. I would like to look through a set of instructions to see where I went wrong.

**Reply by rijkvanzanten** [reply]
What is the relationship you're using right now for this field?

**Reply by limegorilla** [reply]
@rijkvanzanten sorry to reopen this, running into the same issue.
What I've got is a documentation system - something like this

```
documentation/
  | - Top level subdirectory/
  |    | - Example doc
  |    | - Example subsubdirectory/
  |    |    | - Example doc
  | - Example doc

```
What I was trying to do is use a separate collection for the names of the parents - obviously then came up with the error that 'The tree view interface only works for recursive relationships'

Is this the best way to organise this? In my head it's a bit complex for end users as there is no separation in the back end, but it is recursive (something I do want) and allows for a child to have multiple parents, and for each parent to have it's own page.

**Reply by rijkvanzanten** [reply]
There's no way to achieve that in the current tree-view o2m interface. That one only works recursively on a single collection. You could achieve this with 2 collections with a simple m2o in between, but that wouldn't let you browse the items in a tree view currently
